tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -5213,10 +5216,12 @@ void CGMSHLSLRuntime::EmitHLSLRootSignature(CodeGenFunction &CGF,
       ReportHLSLRootSigError(Diags, SLoc,
         (char *)pErrors->GetBufferPointer(), pErrors->GetBufferSize());
       hlsl::DeleteRootSignature(D);
-    }
-    else {
-      llvm::Module *pModule = Fn->getParent();
-      pModule->GetHLModule().GetRootSignature().Assign(D, pSignature);
+    } else {
+      // Only save root signature for entry function.
+      if (Fn == EntryFunc)
```

Comment:
Could we avoid parsing root signatures that are not on the entry function, rather than parsing, then throwing them away?